src/tools/r2rdump/GCSlotTable.cs

```diff
@@ -0,0 +1,208 @@
+                DecodeStackSlots(image, machine, gcInfoTypes, NumUntracked, ref bitOffset);
+            }
+        }
+
+        public override string ToString()
+        {
+            StringBuilder sb = new StringBuilder();
+            string tab2 = new string(' ', 8);
+            string tab3 = new string(' ', 12);
+
+            sb.AppendLine($"{tab2}NumSlots({NumSlots}) = NumRegisters({NumRegisters}) + NumStackSlots({NumStackSlots}) + NumUntracked({NumUntracked})");
+            sb.AppendLine($"{tab2}GcSlots:");
+            sb.AppendLine($"{tab3}-------------------------");
+            foreach (GcSlot slot in GcSlots)
+            {
+                sb.Append(slot.ToString());
+                sb.AppendLine($"{tab3}-------------------------");
+            }
+
+            return sb.ToString();
+        }
+
+        private int DenormalizeStackSlot(Machine target, int x)
+        {
+            switch (target)
+            {
+                case Machine.Amd64:
+                    return (x << 3);
+                case Machine.Arm:
+                    return (x << 2);
+                case Machine.Arm64:
+                    return (x << 3);
+            }
+            return x;
+        }
+
+        private void DecodeRegisters(byte[] image, GcInfoTypes gcInfoTypes, ref int bitOffset)
+        {
+            // We certainly predecode the first register
+            uint regNum = NativeReader.DecodeVarLengthUnsigned(image, gcInfoTypes.REGISTER_ENCBASE, ref bitOffset);
+            GcSlotFlags flags = (GcSlotFlags)NativeReader.ReadBits(image, 2, ref bitOffset);
+            GcSlots.Add(new GcSlot((int)regNum, null, flags));
+
+            for (int i = 1; i < NumRegisters && i < gcInfoTypes.MAX_PREDECODED_SLOTS; i++)
+            {
+                if ((uint)flags != 0)
+                {
+                    regNum = NativeReader.DecodeVarLengthUnsigned(image, gcInfoTypes.REGISTER_ENCBASE, ref bitOffset);
+                    flags = (GcSlotFlags)NativeReader.ReadBits(image, 2, ref bitOffset);
+                }
+                else
+                {
+                    uint regDelta = NativeReader.DecodeVarLengthUnsigned(image, gcInfoTypes.REGISTER_DELTA_ENCBASE, ref bitOffset) + 1;
+                    regNum += regDelta;
+                }
+                GcSlots.Add(new GcSlot((int)regNum, null, flags));
+            }
+        }
+
+        private void DecodeStackSlots(byte[] image, Machine machine, GcInfoTypes gcInfoTypes, uint nSlots, ref int bitOffset)
+        {
+            // We have stack slots left and more room to predecode
+            GcStackSlotBase spBase = (GcStackSlotBase)NativeReader.ReadBits(image, 2, ref bitOffset);
+            int normSpOffset = NativeReader.DecodeVarLengthSigned(image, gcInfoTypes.STACK_SLOT_ENCBASE, ref bitOffset);
+            int spOffset = DenormalizeStackSlot(machine, normSpOffset);
+            GcSlotFlags flags = (GcSlotFlags)NativeReader.ReadBits(image, 2, ref bitOffset);
```

Comment:
When the flag is GC_SLOT_UNTRACKED, the gcinfoencoder.cpp writes the flag with value 4 into 2 bits, so it becomes 0, so my program reads 0 instead of the correct value 4. How do we fix this?